src/BuildScriptGeneratorCli/Commands/BuildCommand.cs

```diff
@@ -339,31 +336,99 @@ internal override bool IsValidInput(IServiceProvider serviceProvider, IConsole c
-                sourceDir: SourceDir,
-                destinationDir: DestinationDir,
-                intermediateDir: IntermediateDir,
-                manifestDir: ManifestDir,
-                platform: PlatformName,
-                platformVersion: PlatformVersion,
-                shouldPackage: ShouldPackage,
-                requiredOsPackages: string.IsNullOrWhiteSpace(OsRequirements) ? null : OsRequirements.Split(','),
-                scriptOnly: false,
-                properties: Properties);
-        }
-
         internal override IServiceProvider GetServiceProvider(IConsole console)
         {
+            // Gather all the values supplied by the user in command line
+            SourceDir = string.IsNullOrEmpty(SourceDir) ?
+                Directory.GetCurrentDirectory() : Path.GetFullPath(SourceDir);
+            ManifestDir = string.IsNullOrEmpty(ManifestDir) ? null : Path.GetFullPath(ManifestDir);
+            IntermediateDir = string.IsNullOrEmpty(IntermediateDir) ? null : Path.GetFullPath(IntermediateDir);
+            DestinationDir = string.IsNullOrEmpty(DestinationDir) ? null : Path.GetFullPath(DestinationDir);
+            var buildProperties = ProcessProperties(Properties);
+
+            // NOTE: Order of the following is important. So a command line provided value has higher precedence
+            // than the value provided in a configuration file of the repo.
+            var config = new ConfigurationBuilder()
+                .AddIniFile(Path.Combine(SourceDir, Constants.BuildEnvironmentFileName), optional: true)
+                .AddEnvironmentVariables()
+                .Add(GetCommandLineConfigSource(buildProperties))
+                .Build();
+
             // Override the GetServiceProvider() call in CommandBase to pass the IConsole instance to
             // ServiceProviderBuilder and allow for writing to the console if needed during this command.
             var serviceProviderBuilder = new ServiceProviderBuilder(LogFilePath, console)
-                .ConfigureScriptGenerationOptions(opts => ConfigureBuildScriptGeneratorOptions(opts));
+                .ConfigureServices(services =>
+                {
+                    // Configure Options related services
+                    // We first add IConfiguration to DI so that option services like
+                    // `DotNetCoreScriptGeneratorOptionsSetup` services can get it through DI and read from the config
+                    // and set the options.
+                    services
+                        .AddSingleton<IConfiguration>(config)
+                        .AddOptionsServices()
+                        .Configure<BuildScriptGeneratorOptions>(options =>
+                        {
+                            // These values are not retrieve through the 'config' api since we do not expect
+                            // them to be provided by an end user.
+                            options.SourceDir = SourceDir;
```

Comment:
curious what are the difference between these fields with the one above which we get from users?